### PR TITLE
fix: Checks For Cast Devices

### PR DESCRIPTION
**Pull request by redreceipt**
## DESCRIPTION

Screen Shot 2020-10-06 at 8 17 36 AM

(This image has a random icon where airplay should be because it doesn't show up in a sim. Just to show that Google Cast is not there.)

### What does this PR do, or why is it needed?

This will check if any cast devices are available and then set local state accordingly so there's not an empty hole where Google Cast should be.

## TODO:

- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] Closes [tag-issues-here]
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [ ] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by redreceipt** [comment]
just waiting to test this. need to add a video to the CMS and prob boot a real device

File: packages/apollos-ui-media-player/src/MediaPlayer/GoogleCastController.js [hunk 1]
```diff
@@ -40,6 +41,10 @@ class Controller extends React.Component {
   componentDidMount() {
     // get Google Cast state on mount
     GoogleCast.getCastState().then((state) => {
+      const noDevices = state === 'NoDevicesAvailable';
+      this.props.client.mutate({
+        mutation: gql`mutation {mediaPlayerUpdateState(isCastAvailable: ${!noDevices}) @client }`,
```
Comment (vinnyjth) [inline packages/apollos-ui-media-player/src/MediaPlayer/GoogleCastController.js:46]: We should probably move the mutation into its own `const` to match how we handle the other mutations in this file.

File: packages/apollos-ui-media-player/src/MediaPlayer/FullscreenPlayer.js [hunk 2]
```diff
@@ -234,7 +238,7 @@ class FullscreenPlayer extends PureComponent {
             showAudioToggleControl={this.props.showAudioToggleControl}
             showVideoToggleControl={this.props.showVideoToggleControl}
             airPlayEnabled={this.props.airPlayEnabled}
-            googleCastEnabled={this.props.googleCastEnabled}
+            googleCastEnabled={this.props.googleCastEnabled && isCastAvailable}
```
Comment (vinnyjth) [inline packages/apollos-ui-media-player/src/MediaPlayer/FullscreenPlayer.js:241]: 👍 👍

**Review by vinnyjth — APPROVED** [review]

File: packages/apollos-ui-media-player/src/MediaPlayer/GoogleCastController.js (hunk 1 above)
Comment (redreceipt) [reply packages/apollos-ui-media-player/src/MediaPlayer/GoogleCastController.js:46]: the only reason I didn't is that all this is changing soon anyway (hopefully 🤞) plus how we do mutations and queries on this package is kind of an antipattern. logic is spread out and very difficult to find. to match how we did it before, I would need a `CAST_NOT_AVAILABLE` and `CAST_AVAILABLE` mutation and would have made this line and logic longer instead of just swapping the variable.

So I did give it some thought but if you'd like me to match anyway, no problem, I can swap it out.

Comment (vinnyjth) [reply packages/apollos-ui-media-player/src/MediaPlayer/GoogleCastController.js:46]: Ah, I see what you're saying. Would it be possible to have a `variable` in this mutation to get rid of the need to have two mutations? 

something like 

```
mutation updateIsCastAvailable($isCastAvailable: Boolean){
  mediaPlayerUpdateState(isCastAvailable: $isCastAvailable) @client
}
```

Comment (vinnyjth) [reply packages/apollos-ui-media-player/src/MediaPlayer/GoogleCastController.js:46]: The resolver would see the same mutation/value with either syntax.

Comment (redreceipt) [reply packages/apollos-ui-media-player/src/MediaPlayer/GoogleCastController.js:46]: yep I could do that

**Comment by redreceipt** [comment]
alright @vinnyjth check it out now

**Comment by vinnyjth** [comment]
@redreceipt Looks good to me! Fix that linter error and we should be good to merge.
